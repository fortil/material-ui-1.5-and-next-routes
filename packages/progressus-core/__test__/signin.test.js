/* eslint-env jest */
/* 
  @By William Penagos billalpeza@gmail.com
*/
import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import App from '../signin'
const Sing = App()

describe('Functions SIGN testing', () => {
  const sing = jest.fn()
  const app = shallow(<Sing register={sing} />).dive()
  const state = app.state()
  const keys = Object.keys(state)
  const email = 'william.penagos@sourcemeridian.com'
  const emailWrong = 'william.penagos'
  const pwd = '123sourcemeridian.com'
  const pwdWrong = '123'

  const fields = [
    { name: 'William' },
    { lastname: 'Penagos' },
    { password: '123sourcemeridian.com' },
    { passwordconfirmed: '123sourcemeridian.com' },
    { type: 'user' },
    { email: 'william.penagos@sourcemeridian.com' }
  ]

  it('Register a user correctly', () => {
    keys.forEach(key => {
      const obj = state[key]
      let value
      if (obj.type === 'email') {
        value = email
      } else if (obj.type === 'password') {
        value = pwd
      } else if (obj.type === 'radio') {
        value = 'user'
      } else {
        if (obj.name === 'name') {
          value = 'William'
        } else {
          value = 'Penagos'
        }
      }
      app.find(`#${key}`).simulate('change', { target: { value } })
      expect(app.state(key)).toEqual(expect.objectContaining({ value }))
    })
    app.find('#register').simulate('click')
    expect(sing).toBeCalledWith(fields)
  })
  it('Put bad password', () => {
    app.find(`#password`).simulate('change', { target: { value: pwd } })
    app.find(`#passwordconfirmed`).simulate('change', { target: { value: pwdWrong } })
    expect(app.state(`passwordconfirmed`)).toEqual(expect.objectContaining({
      error: true,
      errorText: 'El valor no es igual',
    }))
  })
  it('Put bad email', () => {
    app.find(`#email`).simulate('change', { target: { value: emailWrong } })
    expect(app.state(`email`)).toEqual(expect.objectContaining({
      error: true,
      errorText: 'Ingrese un email vÃ¡lido',
    }))
  })
})

describe('With Snapshot Testing', () => {
  it('Snapshot of Registry', () => {
    const component = renderer.create(<Sing />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
