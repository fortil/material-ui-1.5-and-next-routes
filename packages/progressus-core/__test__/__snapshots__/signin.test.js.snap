// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Snapshot of Registry 1`] = `
<div
  className="MuiPaper-root-13 MuiPaper-elevation4-19 MuiPaper-rounded-14 Registry-paper-1"
>
  <div
    className="MuiGrid-container-40 MuiGrid-align-items-xs-center-48 MuiGrid-justify-xs-center-57"
  >
    <div
      className="MuiGrid-item-41 MuiGrid-grid-md-4-100 Registry-image-3"
    >
      <img
        alt="PROGRESUS"
        src="http://lorempixel.com/400/200"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="MuiGrid-item-41 MuiGrid-grid-md-8-104"
    >
      <div
        className="MuiGrid-container-40 MuiGrid-direction-xs-column-43 MuiGrid-align-items-xs-center-48 MuiGrid-justify-xs-center-57"
      >
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-12-80 Registry-titleContainer-4"
        >
          <h2
            className="MuiTypography-root-137 MuiTypography-title-143 Registry-title-5"
          >
            Crea tu cuenta
          </h2>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-12-80"
        >
          <div
            className="MuiFormControl-root-162"
            errorText=""
          >
            <label
              className="MuiFormLabel-root-171 MuiInputLabel-root-166 MuiInputLabel-formControl-167 MuiInputLabel-animated-170"
              data-shrink={false}
              htmlFor="name"
            >
              Nombres
            </label>
            <div
              className="MuiInput-root-178 MuiInput-formControl-179 MuiInput-underline-182"
            >
              <input
                aria-invalid={false}
                className="MuiInput-input-186"
                disabled={false}
                id="name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-12-80"
        >
          <div
            className="MuiFormControl-root-162"
            errorText=""
          >
            <label
              className="MuiFormLabel-root-171 MuiInputLabel-root-166 MuiInputLabel-formControl-167 MuiInputLabel-animated-170"
              data-shrink={false}
              htmlFor="lastname"
            >
              Apellidos
            </label>
            <div
              className="MuiInput-root-178 MuiInput-formControl-179 MuiInput-underline-182"
            >
              <input
                aria-invalid={false}
                className="MuiInput-input-186"
                disabled={false}
                id="lastname"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-12-80"
        >
          <div
            className="MuiFormControl-root-162"
            errorText=""
          >
            <label
              className="MuiFormLabel-root-171 MuiInputLabel-root-166 MuiInputLabel-formControl-167 MuiInputLabel-animated-170"
              data-shrink={false}
              htmlFor="password"
            >
              Contraseña
            </label>
            <div
              className="MuiInput-root-178 MuiInput-formControl-179 MuiInput-underline-182"
            >
              <input
                aria-invalid={false}
                className="MuiInput-input-186 MuiInput-inputType-189"
                disabled={false}
                id="password"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-12-80"
        >
          <div
            className="MuiFormControl-root-162"
            errorText=""
          >
            <label
              className="MuiFormLabel-root-171 MuiInputLabel-root-166 MuiInputLabel-formControl-167 MuiInputLabel-animated-170"
              data-shrink={false}
              htmlFor="passwordconfirmed"
            >
              Confirmar contraseña
            </label>
            <div
              className="MuiInput-root-178 MuiInput-formControl-179 MuiInput-underline-182"
            >
              <input
                aria-invalid={false}
                className="MuiInput-input-186 MuiInput-inputType-189"
                disabled={false}
                id="passwordconfirmed"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
              />
            </div>
          </div>
        </div>
        <fieldset
          className="MuiFormControl-root-162"
        >
          <div
            aria-label="type"
            className="MuiFormGroup-root-191 MuiFormGroup-row-192"
            id="type"
            role="radiogroup"
          >
            <label
              className="MuiFormControlLabel-root-193"
            >
              <span
                className="MuiButtonBase-root-212 MuiIconButton-root-206 MuiSwitchBase-root-202 MuiRadio-root-197 MuiRadio-colorSecondary-201 MuiSwitchBase-checked-203 MuiRadio-checked-198"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label-211"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-215"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <input
                    checked={true}
                    className="MuiSwitchBase-input-205"
                    disabled={false}
                    name="type"
                    onChange={[Function]}
                    type="radio"
                    value="user"
                  />
                </span>
                <span
                  className="MuiTouchRipple-root-222"
                />
              </span>
              <span
                className="MuiTypography-root-137 MuiTypography-body1-146 MuiFormControlLabel-label-196"
              >
                Consultante
              </span>
            </label>
            <label
              className="MuiFormControlLabel-root-193"
            >
              <span
                className="MuiButtonBase-root-212 MuiIconButton-root-206 MuiSwitchBase-root-202 MuiRadio-root-197 MuiRadio-colorSecondary-201"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label-211"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-215"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <input
                    checked={false}
                    className="MuiSwitchBase-input-205"
                    disabled={false}
                    name="type"
                    onChange={[Function]}
                    type="radio"
                    value="doctor"
                  />
                </span>
                <span
                  className="MuiTouchRipple-root-222"
                />
              </span>
              <span
                className="MuiTypography-root-137 MuiTypography-body1-146 MuiFormControlLabel-label-196"
              >
                Psicólogo
              </span>
            </label>
          </div>
        </fieldset>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-12-80"
        >
          <div
            className="MuiFormControl-root-162"
            errorText=""
          >
            <label
              className="MuiFormLabel-root-171 MuiInputLabel-root-166 MuiInputLabel-formControl-167 MuiInputLabel-animated-170"
              data-shrink={false}
              htmlFor="email"
            >
              Email
            </label>
            <div
              className="MuiInput-root-178 MuiInput-formControl-179 MuiInput-underline-182"
            >
              <input
                aria-invalid={false}
                className="MuiInput-input-186 MuiInput-inputType-189"
                disabled={false}
                id="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="email"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-12-80 Registry-buttonContainer-11"
        >
          <button
            className="MuiButtonBase-root-212 MuiButton-root-229 MuiButton-contained-240 MuiButton-containedPrimary-241 MuiButton-raised-243 MuiButton-raisedPrimary-244 Registry-button-12"
            disabled={false}
            id="register"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-230"
            >
              Registrar
            </span>
            <span
              className="MuiTouchRipple-root-222"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
